src/ejson2/EjsonEncoder.ts

```diff
@@ -0,0 +1,241 @@
+import {
+  BsonBinary,
+  BsonDbPointer,
+  BsonDecimal128,
+  BsonFloat,
+  BsonInt32,
+  BsonInt64,
+  BsonJavascriptCode,
+  BsonJavascriptCodeWithScope,
+  BsonMaxKey,
+  BsonMinKey,
+  BsonObjectId,
+  BsonSymbol,
+  BsonTimestamp,
+} from '../bson/values';
+import {fromBase64} from '@jsonjoy.com/base64';
+
+export interface EjsonEncoderOptions {
+  /** Use canonical format (preserves all type information) or relaxed format (more readable) */
+  canonical?: boolean;
+}
+
+export class EjsonEncoder {
+  constructor(private options: EjsonEncoderOptions = {}) {}
+
+  public encode(value: unknown): string {
+    return JSON.stringify(this.transform(value));
```

Comment:
Do not use `JSON.stringify()` and do no "transform" into the other form. Instead write to the `Writer` directly, see how it is done in `JsonEncoder`.